src/js/experiments/snippetFeedback.js

```diff
@@ -0,0 +1,250 @@
+          textForm.value = textForm.value.substring(0, this.maxCharCount - 1);
+        }
+
+        // every time the user types we need to re-render the dom content manually.
+        this.renderCharCount(this.currCharCount);
+      });
+
+      const sendButton = this.makeElement({
+        kind: 'button',
+        className: 'text-btn',
+        text: 'Send',
+        onClick: () => {
+          this.trackFormSubmission(textForm.value);
+          this.wasThisHelpfulContainer.innerHTML =
+            '<div>Thank you for your feedback!</div>';
+        },
+      });
+
+      // build the dom nodes.
+      container.appendChild(prompt);
+      container.appendChild(textForm);
+      bottomRow.appendChild(this.currCharCountElement);
+      bottomRow.appendChild(sendButton);
+      container.appendChild(bottomRow);
+      this.feedbackForm = container;
+      this.wasThisHelpfulContainer.appendChild(container);
+
+      // Here we setup the document event listeners that handle hiding the form
+      // when the user clicks outside of it.
+      //
+      // Why do we use a setTimeout with a timer of 0?
+      // -> Because there is a race condition with the adding of the click
+      // listeners to the document that immediately close the form when it opens
+      //
+      // Setting it to 0 causes the inner callback to be run asynchronously, adding it to the eventqueue
+      // so that it executes after the form being opened.
+      setTimeout(() => {
+        const outsideClickListener = (event) => {
+          if (!this.feedbackForm.contains(event.target)) {
+            this.wasThisHelpfulContainer.removeChild(this.feedbackForm);
+            this.feedbackForm = null;
+            removeClickListener();
+          }
+        };
+
+        const removeClickListener = () => {
+          document.removeEventListener('click', outsideClickListener);
+        };
+
+        document.addEventListener('click', outsideClickListener);
+      }, 0);
+    }
+  }
+
+  /**
+   * Render the char count and set it's result on the respective element.
+   * */
+  renderCharCount(charCount) {
+    let charCountLimited =
+      charCount > this.maxCharCount ? this.maxCharCount : charCount;
+    const el = `
+        <div class="charCountRow">
+          <span>${charCountLimited} / ${this.maxCharCount}</span>
+        </div>`;
+    this.currCharCountElement.innerHTML = el;
+  }
+
+  // -- Trackers --
+  // These are responsible for sending data based on user input
+  // to amplitude.
+
+  /**
+   * trackYesOrNoButton sends a trackAction to aplitude.
+   * actions are: "docs-snippet-helpful-no"  and "docs-snippet-helpful-yes"
+   * */
+  trackYesOrNoButton(yesOrNoString) {
+    window.AnalyticsClient.trackAction(
+      `docs-snippet-helpful-${yesOrNoString}`,
+      {
+        originatingSnippet: this.snippetElement.textContent,
+        location: window.location.pathname,
+        timeOfButtonClick: Date.now(),
```

Comment:
not sure how this is done elsewhere but it's probably better for chart-making purposes and readability to convert this to an ISO string.

Reply:
Done.